site/src/components/WorkspaceScheduleForm/WorkspaceScheduleForm.tsx

```diff
@@ -262,6 +278,24 @@ export const WorkspaceScheduleForm: FC<WorkspaceScheduleFormProps> = ({
   )
 }
 
+const ttlShutdownAt = (now: dayjs.Dayjs, workspace: Workspace, tz: string, newTTL: number): string => {
+  if (workspace.latest_build.transition !== "start") {
+    return Language.ttlHelperText
+  }
+  if (newTTL === 0) {
+    return Language.ttlCausesNoShutdownHelperText
+  }
+  const newDeadline = dayjs(workspace.latest_build.updated_at).add(newTTL, "hour")
+  if (newDeadline.isBefore(now)) {
+    return `⚠️ ${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownImmediately} ⚠️`
+  }
+  if (newDeadline.isBefore(now.add(30, "minute"))) {
+    return `⚠️ ${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownSoon} ⚠️`
+  }
+  const newDeadlineString = newDeadline.tz(tz).format("hh:mm A z")
+  return `${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownAt} ${newDeadlineString}.`
```

Comment:
NIT: would making the Language value a function alleviate the need for some of these keys? Example:
````
const Language = {
  ttlCausesShutdownAtText: (newDeadlineString: string): string => {
    return `Your workspace will shut down at ${newDeadlineString}`
  },
}
````

Reply:
I guess it would, but it would make the `Language` portion more complex. I feel like we'd want to keep `Language` as a simple lookup table and keep the complexity close to where it's used. I'm fine with going either way though if folks have strong opinions about it!

Reply:
No strong opinions here - that sounds good to me!